server/protocols/webadmin/cli/james_cli

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python3
+
+import click
+import json
+import requests as req
+import sys
+
+
+def print_json(obj):
+    parsed = json.loads(obj.text)
+    print(json.dumps(parsed, indent=4, sort_keys=True))
+
+
+@click.group()
+@click.option("--ip", default="127.0.0.1", show_default=True, help="Set ip address")
+@click.option("--port", default="8000", show_default=True, help="Set port")
+@click.pass_context
+def main(ctx, ip, port):
+    """Run './james_cli COMMAND --help' for more information on a command."""
+    path = "http://" + ip + ":" + port
+    ctx.obj = {
+        "path": path
+    }
+
+
+# Mapping routes implemantation ---------------------------------------
+@main.group()
+def mapping():
+    """Mapping routes implementation"""
+    pass
+
+
+@mapping.command(name="list")
+@click.pass_context
+def get_all_mapping(ctx):
+    """List all mappings"""
+    res = req.get(ctx.obj["path"] + "/mappings")
+    if res.status_code == 200:
+        print_json(res)
+    elif res.status_code == 401:
+        print("Authentication error.")
+    elif res.status_code == 500:
+        print("Internal server error!")
+    else:
+        print("Unhandled response received from the server.")
+        print("Status: " + str(res.status_code))
+        print_json(res)
+
+# Domain routes implemantation ---------------------------------------
+@main.group()
+def domain():
+    """Domain routes implementation"""
+    pass
+
+
+@domain.command(name="list")
+@click.pass_context
+def get_all_domains(ctx):
+    """Show all available domains"""
+    res = req.get(ctx.obj["path"] + "/domains")
+
+    if res.status_code == 200:
```

Comment:
are you going to repeat the error handling code for all actions?

Reply:
In some cases, it has the same codes but it is not always the same. The repeated part is for the unhandled response so I have written a function for that in the latest commit.